projectgenerator/gui/import_data.py

```diff
@@ -392,7 +391,9 @@ def help_requested(self):
                 "https://opengisch.github.io/projectgenerator/docs/user-guide.html#import-an-interlis-transfer-file-xtf")
 
     def advance_progress_bar_by_text(self, text):
-        if text.strip() == 'Info: compile models…':
+        if text.strip() == 'Info: compile models...':
             self.progress_bar.setValue(50)
-        elif text.strip() == 'Info: create table structure…':
+            QCoreApplication.processEvents()
+        elif text.strip() == 'Info: create table structure...':
             self.progress_bar.setValue(75)
+            QCoreApplication.processEvents()
```

Comment:
Does that really fix the crash? Or just reduce the risk?
If it helps something, I am ok to merge this. But I think we should rather send this to a background thread and do this properly (sooner or later).

Reply:
I think it is fair to say that it fixes the crash because instead of calling `QCoreApplication.processEvents()` hundreds (potentially thousands) of times, we now just call it twice at the most.

Reply:
No criticism, it's a good move in the right direction, but it's still more a band aid than a real fix :)

Reply:
... as we actually loose real time progress reporting

Reply:
Could you please tell me why? I still don't get it.

Reply:
Oh, the message log is being updated in a normal way for the user, it doesn't hang or something.

Reply:
Two things:

 * we still have `processEvents()` calls in there, so we can still have crashes
 * we only update the log upon certain events instead of whenever we get new text to show, which can make user wonder if the process is still alive

Reply:
Hmm... might be that Qt fixed a couple of things. I think mac is still affected by the no update. And there's still a risk of crashes, it's just decreased a bit (not sure if it decreases by the number of calls to `processEvents()` or by the amount of time between the last "real" event loop and the last `processEvents()` call though)